plugins/woocommerce/src/Admin/API/Reports/DataStore.php

```diff
@@ -177,10 +218,26 @@ protected function get_cache_key( $params ) {
 	 * @return mixed
 	 */
 	protected function get_cached_data( $cache_key ) {
-		if ( $this->should_use_cache() ) {
-			return Cache::get( $cache_key );
+		if ( true === $this->debug_cache ) {
+			$this->debug_cache_data['should_use_cache']    = $this->should_use_cache();
+			$this->debug_cache_data['force_cache_refresh'] = $this->force_cache_refresh;
+			$this->debug_cache_data['cache_hit']           = false;
+		}
+
+		if ( $this->should_use_cache() && false === $this->force_cache_refresh ) {
+			$cached_data = Cache::get( $cache_key );
+
+			$cache_hit = ! ! $cached_data;
```

Comment:
Sometimes this might return a false negative if an empty array is returned.  For example, if we hit the `/wp-json/wc-analytics/reports/orders` endpoint and no orders exist in a period, an empty array will be returned from the cache.

Reply:
🤔 I noticed that some of the reports do this with an empty result. I guess an empty array would be falsey in this case. Maybe `! empty()` would be a better test here?

Sidenote, but at least in the case of `reports/orders`, it actually [returns early](https://github.com/woocommerce/woocommerce/blob/c3fd7134cce2bd215f2e4473af633d180be2a881/plugins/woocommerce/src/Admin/API/Reports/Orders/DataStore.php#L291-L293) if there's no result, so nothing is getting cached, and it truly is not a cache hit. Which seems... less than ideal, but probably beyond the scope of this PR.

Reply:
> Maybe ! empty() would be a better test here?

Er, I guess it needs to be `false !== $cached_data` so that a cached empty array would still count as a hit.